### PR TITLE
[Serverless] Update java integration tests to use latest layers.

### PR DESCRIPTION
**Pull request by purple4reina**
### What does this PR do?

Updates the java serverless integration tests to use the latest version of the java tracing layer.

### Motivation

Old versions of the layer have a bug.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/integration/serverless/snapshots/error-java [hunk 1]
```diff
@@ -99,6 +99,54 @@
...
+      "runtime:java8.al2",
+      "service:integration-tests-service",
+      "taga:valuea",
+      "tagb:valueb",
+      "tagc:valuec",
+      "tagd:valued",
+      "version:integration-tests-version"
+    ],
+    "dogsketches": []
+  },
```
Comment (purple4reina) [inline test/integration/serverless/snapshots/error-java:149]: Now that the integration test is using universal instrumentation, we are seeing these error metrics being created.

**Comment by purple4reina** [comment]
Updating to the newest version of the java tracing layer uncovered a new issue. It seems the newest version of the tracing layer is double wrapping the lambda handler. See https://datadoghq.atlassian.net/browse/SLS-3586.

File: test/integration/serverless/snapshots/log-java [hunk 2]
```diff
@@ -194,6 +194,51 @@
     "ddsource": "lambda",
     "ddtags": "account_id:425362996713,architecture:XXX,aws_account:425362996713,dd_extension_version:123,env:integration-tests-env,function_arn:arn:aws:lambda:eu-west-1:425362996713:function:integration-tests-extension-XXXXXX-log-java,functionname:integration-tests-extension-XXXXXX-log-java,memorysize:1024,region:eu-west-1,resource:integration-tests-extension-XXXXXX-log-java,runtime:java8.al2,service:integration-tests-service,taga:valuea,tagb:valueb,tagc:valuec,tagd:valued,version:integration-tests-version"
   },
+  {
+    "message": {
+      "message": "[dd.trace TIMESTAMP +0000] [main] ERROR datadog.trace.lambda.LambdaHandler - could not notify the extension as the lambda span is null or no sampling priority has been found\n",
```
Comment (purple4reina) [inline test/integration/serverless/snapshots/log-java:199]: This is an indication of a real issue with the newest version of the tracing layer. It will be addressed in a separate PR.

**Review by duncanista — APPROVED** [review]
LGTM! Thank you so much Rey!
